docs/automation/mobile.md

```diff
@@ -105,27 +121,130 @@ We could provide any Appium capabilty in **config.properties** file using capabi
-capabilities.platformName=IOS
+capabilities.udid=D85FF1CD-D95F-4B78-A007-77A11EBD3ABB
+capabilities.newCommandTimeout=1800
 #=====================================================#
 ```
+
+<b>Note:</b>
+To get device’s unique id (UDID) for emulator command “xcrun simctl list” is useful. It will show the list of simulators with there names, iOS versions, udids and statuses (shutdown or booted). 
+To get UDID from real device you have to plug your phone into a computer and copy it out of iTunes:
+1) Launch iTunes and connect your iPhone.
+2) In the right panel, locate the information about your iPhone, including its name, capacity, software version, serial number, and phone number.
+3) Reveal the Identifier by clicking on Serial Number.
+
+### Implementation of Page Objects:
+The main idea is the same as in [web-testing](http://qaprosoft.github.io/carina/automation/web/#implementation-of-page-objects) except that css isn’t used in Mobile. 
+ExtendedWebElement(Carina’s implementation of WebElement) is used instead of WebElement and it has methods similar to those that WebElement has but more reliable and convenient.
+<b>Important</b>:
+ - Page should extends com.qaprosoft.carina.core.gui.AbstractPage
+ - Use com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement instead of Selenium WebElement
+ - Locate Page Object classes in src/main/java package
+
+### How to find locators for Android application
+To obtain locaters of web elements from Android app different programs are used, such as latest Appium itself and convenient Android SDK tool: uiautomatorviewer. It’s localed in $ANDROID_HOME/tools/bin and could be launched using command line  (./uiautomatorviewer).
+Example:
+```
+ @FindBy(xpath = "//*[@resource-id='name_input']")
+ private ExtendedWebElement input;
+```
+
+
+### How to find locators for iOS application
+To obtain locaters of web elements from iOS app different programs are used, such as GUI Appium itself and convenient [Macaca App Inspector](https://macacajs.github.io/app-inspector/).
+To faster work @Predicate is used. With predicates complicate “xpaths” can’t be used, but “id”, “name”, “className” and etc. search are sufficient. 
+Example:
+```
+@FindBy(xpath = "name = 'DONE'")
+@Predicate
+protected ExtendedWebElement doneButton;
+```
+Another possibility to find element is to use @ClassShain annotation.
+Example:
+```
+@FindBy(xpath = "**/XCUIElementTypeStaticText[`name=='Developer'`]")
+@ClassChain
+protected ExtendedWebElement developerText;
+```
+
+### Implementation of tests
+Carina framework uses TestNG for test organization. In general test represents manipulation with Page Objects and additional validations of UI events. Here is sample test implementation:
+
+```
+public class SampleTest extends AbstractTest {
+	
+	String name = "My name";
+	String carName = "Mercedes";
+	
+
+    @Test()
+    public void sendName() {
+    	FirstPage  firstPage = new FirstPage(getDriver());
```

Comment:
bad practice to declare all page objects in the beginning of the test. for documentation we should provide better example.